devicetypes/smartthings/fibaro-rgbw-controller.src/fibaro-rgbw-controller.groovy

```diff
@@ -26,562 +26,236 @@
-    }
-    result
+def zwaveEvent(physicalgraph.zwave.commands.switchmultilevelv3.SwitchMultilevelReport cmd) {
+	dimmerEvents(cmd)
 }
 
-def createEvent(physicalgraph.zwave.commands.switchmultilevelv1.SwitchMultilevelStartLevelChange cmd, Map item1) {
-     []
+def zwaveEvent(physicalgraph.zwave.commands.switchmultilevelv3.SwitchMultilevelSet cmd) {
+	dimmerEvents(cmd)
 }
 
-def createEvent(physicalgraph.zwave.commands.switchmultilevelv1.SwitchMultilevelStopLevelChange cmd, Map item1) {
-     [response(zwave.basicV1.basicGet())]
+private dimmerEvents(physicalgraph.zwave.Command cmd) {
+	def value = (cmd.value ? "on" : "off")
+	def result = [createEvent(name: "switch", value: value)]
+	if (cmd.value && cmd.value <= 100) {
+		result << createEvent(name: "level", value: cmd.value == 99 ? 100 : cmd.value)
+	}
+	return result
 }
 
-def createEvent(physicalgraph.zwave.commands.switchmultilevelv3.SwitchMultilevelSet cmd, Map item1) {
-     def result = doCreateEvent(cmd, item1)
-     for (int i = 0; i < result.size(); i++) {
-          result[i].type = "physical"
-     }
-     result
+def zwaveEvent(physicalgraph.zwave.commands.switchmultilevelv3.SwitchMultilevelStopLevelChange cmd) {
+	response(zwave.basicV1.basicGet())
 }
 
-def createEvent(physicalgraph.zwave.commands.switchmultilevelv1.SwitchMultilevelReport cmd, Map item1) {
-     def result = doCreateEvent(cmd, item1)
-     result[0].descriptionText = "${item1.linkText} is ${item1.value}"
-     result[0].handlerName = cmd.value ? "statusOn" : "statusOff"
-     for (int i = 0; i < result.size(); i++) {
-          result[i].type = "digital"
-     }
-     result
+def zwaveEvent(physicalgraph.zwave.commands.configurationv2.ConfigurationReport cmd, item1) {
+	log.debug "${device.displayName} parameter '${cmd.parameterNumber}' with a byte size of '${cmd.size}' is set to '${cmd.configurationValue}'"
 }
 
-def doCreateEvent(physicalgraph.zwave.Command cmd, Map item1) {
-     def result = [item1]
-
-     item1.name = "switch"
-     item1.value = cmd.value ? "on" : "off"
-     item1.handlerName = item1.value
-     item1.descriptionText = "${item1.linkText} was turned ${item1.value}"
-     item1.canBeCurrentState = true
-     item1.isStateChange = isStateChange(device, item1.name, item1.value)
-     item1.displayed = item1.isStateChange
-
-     if (cmd.value >= 5) {
-          def item2 = new LinkedHashMap(item1)
-          item2.name = "level"
-          item2.value = (cmd.value == 99 ? 100 : cmd.value) as String
-          item2.unit = "%"
-          item2.descriptionText = "${item1.linkText} dimmed ${item2.value} %"
-          item2.canBeCurrentState = true
-          item2.isStateChange = isStateChange(device, item2.name, item2.value)
-          item2.displayed = false
-          result << item2
-     }
-     result
+def handleMeterReport(cmd){
+	if (cmd.meterType == 1) {
+		if (cmd.scale == 0) {
+			createEvent(name: "energy", value: cmd.scaledMeterValue, unit: "kWh")
```

Comment:
Need to convert to double

Reply:
Going to wait and see on this one to see if that bug report was valid.

Reply:
It wasn't, so I'm going to leave this as is.